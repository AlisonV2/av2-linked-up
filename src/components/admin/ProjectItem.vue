<template>
  <div class="card text-center">
    <div class="card-header title">
      {{ project.title }}
    </div>
    <h6 class="card-subtitle mb-2 text-muted" v-if="role === 'client'">{{ project.artistName }}</h6>
    <h6 class="card-subtitle mb-2 text-muted" v-if="role === 'artist'">{{ project.clientName }}</h6>
    <p class="card-text">{{ project.description }}</p>
    <div class="card-footer text-muted">
      {{ createdAt }}
    </div>
  </div>
</template>

<script>
import { format } from 'date-fns';

/**
 * @exports ProjectItem
 * @type {Component}
 * @requires Sentry
 * @vue-data {array} projects
 */
export default {
  name: 'ProjectItem',
  props: ['project', 'time'],
  data() {
    return {
      createdAt: '',
      role: '',
    };
  },
  created() {
    this.formatDate();
    this.getUserRole();
  },
  methods: {
    formatDate() {
      this.createdAt = format(this.time.toDate(), 'dd/MM/yyyy');
    },
    async getUserRole() {
      await this.$store.dispatch('getUserRole').then(() => {
        this.role = this.$store.getters.getUserRole;
      });
    },
  },
};
</script>
