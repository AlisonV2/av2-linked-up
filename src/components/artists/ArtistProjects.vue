<template>
  <div class="row" v-if="projects.length">
    <div class="col-12 col-md-6 col-xl-4" v-for="project in projects" :key="project">
      <router-link :to="{ name: 'ProjectDetails', params: { id: project.id, isArtist: true } }">
        <ProjectItem :project="project"/>
      </router-link>
    </div>
  </div>
</template>

<script>
import ProjectItem from '@/components/admin/ProjectItem';
import * as Sentry from '@sentry/vue';

/**
 * @exports ArtistProjects
 * @type {Component}
 * @requires Sentry
 * @vue-data {array} projects
 */
export default {
  name: 'ArtistProjects',
  components: {
    ProjectItem,
  },
  data() {
    return {
      projects: [],
    };
  },
  created() {
    this.$store
      .dispatch('getArtistProjects')
      .then(() => {
        this.projects = this.$store.getters.getArtistProjects;
      })
      .catch((err) => {
        Sentry.captureException(err);
      });
  },
};
</script>
