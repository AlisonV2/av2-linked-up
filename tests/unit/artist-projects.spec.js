import { shallowMount } from '@vue/test-utils';
import { createStore } from 'vuex';
import ArtistProjects from '@/components/artists/ArtistProjects.vue';
import ProjectItem from '@/components/admin/ProjectItem.vue';

const actions = {
  getArtistProjects: jest.fn(),
};

const getters = {
  getArtistProjects: jest.fn(),
};

const store = createStore({
  actions,
  getters,
});

const config = {
  data() {
    return {
      projects: [],
    };
  },
  global: {
    plugins: [store],
    components: {
        ProjectItem,
    }
  },
};

/**
 * @module ArtistProjectsTest
 */
describe('ArtistProjects.vue', () => {
  it('ArtistProjects Snapshot', () => {
    const component = shallowMount(ArtistProjects, config);
    expect(component.element).toMatchSnapshot();
  });

  it('Check if getArtistProjects action is called', async () => {
    const component = shallowMount(ArtistProjects, config);
    ArtistProjects.created.call(component.vm);
    await component.vm.$nextTick();
    expect(actions.getArtistProjects).toHaveBeenCalled;
  });

  it('Check if getArtistProjects getter is called', async () => {
    const component = shallowMount(ArtistProjects, config);
    ArtistProjects.created.call(component.vm);
    await component.vm.$nextTick();
    expect(getters.getArtistProjects).toHaveBeenCalled;
  });
});
