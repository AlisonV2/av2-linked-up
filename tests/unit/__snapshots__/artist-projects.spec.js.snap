// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistProjects.vue ArtistProjects Snapshot 1`] = `<!--v-if-->`;
